app/(service)/recruit-detail/[shopId]/[recruitId]/_components/recruitCard.tsx

```diff
@@ -0,0 +1,104 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import Image from 'next/image';
+import { useParams } from 'next/navigation';
+
+import Button from '@/components/button/button';
+import PercentageBadge from '@/components/post/percentageBadge';
+import apiClient from '@/libs/axios';
+import calculatePercentage from '@/utils/calculatePercentage';
+import formatDateRange from '@/utils/formatDateRange';
+
+interface FetchData {
+  hourlyPay: 'number';
+  startsAt: 'string';
+  workhour: 'number';
+  description: 'string';
+  closed: 'boolean';
+  shop: {
+    item: {
+      name: 'string';
+      category: 'string';
+      address1: 'string';
+      description: 'string';
+      imageUrl: 'string';
+      originalHourlyPay: 'number';
+    };
+  };
+}
+
+function RecruitCard() {
+  const [recruitData, setRecruitData] = useState<FetchData | null>(null);
+  const { shopId, recruitId } = useParams();
+  // 임시로 쓴 데이터 요청임
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const res = await apiClient.get(`/shops/${shopId}/notices/${recruitId}`);
+        setRecruitData(res.data.item);
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, [shopId, recruitId]);
+  const percentage = calculatePercentage(recruitData?.hourlyPay, recruitData?.shop.item.originalHourlyPay);
```

Comment:
P5: 일반적으로 useEffect가 return 바로 위로 들어간다고 알고 있어서, 이렇게 코드 순서를 바꿔주시는 것도 좋을 것 같습니다. 

```suggestion
  const [recruitData, setRecruitData] = useState<FetchData | null>(null);
  const { shopId, recruitId } = useParams();
  
  const percentage = calculatePercentage(recruitData?.hourlyPay, recruitData?.shop.item.originalHourlyPay);
    
  // 임시로 쓴 데이터 요청임
  useEffect(() => {
    const fetchData = async () => {
      try {
        const res = await apiClient.get(`/shops/${shopId}/notices/${recruitId}`);
        setRecruitData(res.data.item);
      } catch (error) {
        console.error('Error fetching data:', error);
      }
    };

    fetchData();
  }, [shopId, recruitId]);
```

Reply:
확인했습니다